### ISSUE TITLE
Potential for hung steps due to implicit process exit

### ISSUE THREAD

**Issue by jclem** [actions/cache#370]
**tl;dr** When this action's [`run`](https://github.com/actions/cache/blob/eed9cfe64d00dd64cbb36bec915d41dd6fab9f6c/src/restore.ts#L63) function completes, there is a potential for jobs to [hang](https://github.com/android-password-store/Android-Password-Store/runs/868857729?check_suite_focus=true) since we aren't explicitly exiting the process (due to other bugs leaving active event listeners around).

The Node.js runtime keeps track of _active event listeners_. When there are no more active event listeners, the [program exits](https://bytearcher.com/articles/why-does-hello-world-exit-but-http-server-doesnt/). This is why a process that just runs `console.log("Hello, world")` exits when the statement executes, but a process that starts an HTTP server doesn't exit after the server has started.

In order to prevent this action from hanging in the event that there are active event listeners somehow still around once `run()` completes, we should explicitly exit the process:

```javascript
run().then(() => process.exit())
```
